### ISSUE TITLE
[Feature request] vagrant-sshfs support

### ISSUE THREAD

**Issue by vincenzo** [puphpet/puphpet#1946]
I find [vagrant-sshfs](https://github.com/fabiokr/vagrant-sshfs) plugin very useful and I've actually modified my puphpet-built VM to support it to the extent of my personal use case. The following is my diff against `Vagrantfile-local`.

```diff
diff --git a/puphpet/vagrant/Vagrantfile-local b/puphpet/vagrant/Vagrantfile-local
index bb127bb..51fea6b 100644
--- a/puphpet/vagrant/Vagrantfile-local
+++ b/puphpet/vagrant/Vagrantfile-local
@@ -106,6 +106,12 @@ Vagrant.configure('2') do |config|
 
         config.vm.synced_folder "#{folder['source']}", "#{folder['target']}", id: "#{i}",
           rsync__args: rsync_args, rsync__exclude: rsync_exclude, rsync__auto: rsync_auto, type: 'rsync', group: sync_group, owner: sync_owner
+      elsif folder['sync_type'] == 'sshfs'
+        if Vagrant.has_plugin?('vagrant-sshfs')
+          config.sshfs.paths = { "#{folder['target']}" => "#{folder['source']}" }
+        else
+          raise "\n\nPlugin vagrant-sshfs is required for SSHFS shares.\nPlease run 'vagrant plugin install vagrant-sshfs' to install the plugin, then try again.\n\n"
+        end
       elsif data['vm']['chosen_provider'] == 'parallels'
         config.vm.synced_folder "#{folder['source']}", "#{folder['target']}", id: "#{i}",
           group: sync_group, owner: sync_owner, mount_options: ['share']
```

And of course I have `sync_type: sshfs` in my `config.yaml` under the `synced_folder:` section. 

Notice that target and source are inverted for this type of "share mount". This is because the default (guest) behaviour of this plugin is to mount a guest's directory onto the host. And not vice versa. 

I could work on a pull request that provides more thorough changes for `Vagrantfile-local` so it could support `vagrant-sshfs`-specific settings in `config.yaml`, like

```
                sshfs:
                    sshfs_username: ''
                    sshfs_enabled: ''
```

But that's as far as I could go with my knowledge of `puphpet` codebase. I wouldn't know were to go to expose the support for SSHFS on the puphpet's Web UI, neither would I know what to change so that a `config.yaml` could be generated accordingly. But I am happy to be pointed to the right direction.

**Comment by jtreminio** [comment]
Looks like the plugin was abandoned: https://github.com/fabiokr/vagrant-sshfs/issues/38#issuecomment-159902491

Are you sure it works with the latest Vagrant release?

**Comment by vincenzo** [comment]
@jtreminio Yes, it does. Plus, I'm looking to take over maintenance. I'm getting familiar with Ruby and vagrant plugin development first. The plugin is simple enough to maintain anyway. Fabio said he's happy to merge PRs. But I can fork it if he abandons the plugin altogether.

**Comment by vincenzo** [comment]
@jtreminio FYI, the `vagrant-sshfs` has been forked: https://github.com/dustymabe/vagrant-sshfs,  and is now actively maintained by @dustymaybe. So perhaps this could be officially supported now? I keep adding it to my boxes, anyway :)
